From bc164bb2de746010b6a8141356e86562ffd3222c Mon Sep 17 00:00:00 2001
Message-ID: <bc164bb2de746010b6a8141356e86562ffd3222c.1696804840.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1696804840.git.sharlatanus@gmail.com>
References: <cover.1696804840.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 5 Oct 2023 00:57:00 +0100
Subject: [PATCH 06/20] gnu: python-jplephem: Improve package style.

* gnu/packages/astronomy.scm (python-jplephem)[build-system]: Shift to
pyproject-build-system.
[arguments]: Use G-expressions. Simplify 'check phase.
[description]: Extend it with more details.
---
 gnu/packages/astronomy.scm | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index d0aedf3296..335ee6cd41 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -3205,24 +3205,25 @@ (define-public python-jplephem
        (uri (pypi-uri "jplephem" version))
        (sha256
         (base32 "1kvdzwyh0invinvcnwfr14r9zbqzcwkdh0cpzw1m4vkdcm64aqn1"))))
-    (build-system python-build-system)
+    (build-system pyproject-build-system)
     (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key tests? inputs outputs #:allow-other-keys)
-             (when tests?
-               (let ((out (assoc-ref outputs "out")))
-                 (add-installed-pythonpath inputs outputs)
-                 (setenv "PATH" (string-append out "/bin:" (getenv "PATH")))
-                 (invoke "python" "-m" "unittest" "discover" "-s" "test"))))))))
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (replace 'check
+            (lambda* (#:key tests? #:allow-other-keys)
+              (when tests?
+                (invoke "python" "-m" "unittest" "discover" "-s" "test")))))))
     (inputs
      (list python-numpy))
     (home-page "https://github.com/brandon-rhodes/python-jplephem")
     (synopsis "Python version of NASA DE4xx ephemerides")
     (description
-     "The package is a Python implementation of the mathematics that standard
-JPL ephemerides use to predict raw (x,y,z) planetary positions.")
+     "@code{skyfield} computes positions for the stars, planets, and
+satellites in orbit around the Earth.  Its results should agree with the
+positions generated by the United States Naval Observatory and their
+Astronomical Almanac to within 0.0005 arcseconds (half a @emph{mas} or
+milliarcsecond).")
     (license license:expat)))
 
 (define-public python-jwst
-- 
2.41.0

